Content.Server/Clothing/MaskSystem.cs

```diff
@@ -0,0 +1,101 @@
+using Content.Shared.Actions;
+using Content.Shared.Toggleable;
+using Content.Shared.Inventory;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Item;
+using Content.Server.Actions;
+using Content.Server.Atmos.Components;
+using Content.Server.Body.Components;
+using Content.Server.Clothing.Components;
+using Content.Server.Disease.Components;
+using Content.Server.Nutrition.EntitySystems;
+using Content.Server.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Clothing
+{
+    public sealed class MaskSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly InventorySystem _inventorySystem = default!;
+        [Dependency] private readonly ActionsSystem _actionSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MaskComponent, ToggleMaskEvent>(OnToggleMask);
+            SubscribeLocalEvent<MaskComponent, GetItemActionsEvent>(OnGetActions);
+            SubscribeLocalEvent<MaskComponent, GotUnequippedEvent>(OnGotUnequipped);
+        }
+
+        private void OnGetActions(EntityUid uid, MaskComponent component, GetItemActionsEvent args)
+        {
+            if (component.ToggleAction != null)
+                args.Actions.Add(component.ToggleAction);
+        }
+
+        private void OnToggleMask(EntityUid uid, MaskComponent mask, ToggleMaskEvent args)
+        {
+            if (!_inventorySystem.TryGetSlotEntity(args.Performer, "mask", out var existing) || !mask.Owner.Equals(existing))
+                return;
+
+            mask.IsToggled ^= true;
+            _actionSystem.SetToggled(mask.ToggleAction!, mask.IsToggled);
```

Comment:
this will NRE. remove the `!` and handle the null value properly using an if-statement